File: Practice/Practice.java
import java.util.*;
public class Practice {

  List<String> items = new ArrayList<String>();

  public int getClassiness(){ 
    int res = 0;
    if (items.size() > 0) {
      for (int i = 0; i < items.size(); i++){
        if (items.get(i) == "tophat"){
          res += 2;
        }
        if (items.get(i) == "bowtie"){
          res += 4;
        }
        if (items.get(i) == "monocle"){
          res += 5;
        }              
      }
    }
    return res;
  }

  private void addItem(String newItem){
    items.add(newItem);
  }  

  public static void main(String[] args){
    Practice practice = new Practice();
    practice.addItem("tophat");
    practice.addItem("bowtie");
    practice.addItem("jacket");
    practice.addItem("monocle");
    practice.addItem("bowtie");
    System.out.println(practice.items);
    System.out.println(practice.getClassiness());
  }
}
  

File: Practice/Practice.go
package main

import (
	"fmt"
)

func main() {
	//items2 := make([]string, 0)
	c := &Classy{}
	c.addItem("tophat")
	c.addItem("bowtie")
	c.addItem("jacket")
	c.addItem("monocle")
	fmt.Println(c.getClassiness())
	c.addItem("bowtie")
	fmt.Println(c.getClassiness())
}

//Classy is interchangable with "fancy".
//If you add fancy-looking items, you will increase your "classiness".
type Classy struct {
	items []string
}

func (c *Classy) getClassiness() int {
	classinessCounter := 0
	if len(c.items) > 0 {
		for _, item := range c.items { //use := instead of in
			if item == "tophat" {
				classinessCounter += 2
			}
			if item == "bowtie" {
				classinessCounter += 4
			}
			if item == "monocle" {
				classinessCounter += 5
			}
		}
		return classinessCounter
	}
	return classinessCounter
}

func (c *Classy) addItem(newItem string) {
	c.items = append(c.items, newItem)
}
